presto-main/src/main/java/com/facebook/presto/metadata/TableLayoutResult.java

```diff
@@ -56,4 +60,28 @@ public boolean hasAllOutputs(TableScanNode node)
 
         return columns.containsAll(nodeColumnHandles);
     }
+
+    public static TupleDomain<ColumnHandle> computeEnforced(TupleDomain<ColumnHandle> tupleDomain, TupleDomain<ColumnHandle> unenforcedTupleDomain)
+    {
+        if (tupleDomain.isNone()) {
+            checkArgument(unenforcedTupleDomain.isNone() || unenforcedTupleDomain.isAll());
```

Comment:
If the engine requests a connector provides a layout with a domain of "none". The connector can have two possible reactions, either:

1. It can provide an empty table layout, in which case there would be no unenforced domain
2. It cannot. In which case, it will tell the engine to help it filter everything out by providing a "false" unenforced domain.

I had this here as a sanity check to assert one of these two scenarios. It would be invalid argument otherwise. I'll put my answer above as comment in the code.

Reply:
> If the engine requests a connector provides a layout with a domain of "none"

It should not happen. `com.facebook.presto.metadata.MetadataManager#getLayouts` will return an empty list for domain `none`.